File: OpenValidationFramework-Java/src/main/java/org/bag/openvalidation/validator/EnumValidator.java
package org.bag.openvalidation.validator;

import org.bag.model.EnumModel;
import org.bag.model.Enums;
import org.bag.openvalidation.HUMLFramework;

public class EnumValidator implements HUMLFramework.IOpenValidator {
    public HUMLFramework huml;

    private EnumValidator(){

    }

    public static EnumValidator createFirstEnumValidator()
    {
        EnumValidator validator = new EnumValidator();
        validator.huml = new HUMLFramework();

        HUMLFramework.Variable X = validator.huml.CreateVariable("X", (EnumModel model) ->
                validator.huml.FIRST(model.getEnums())

        );

        //HUMLFramework.Variable X = huml.CreateVariable("X", (BooleanModel model) -> huml.FIRST(Y.GetValue(model)));

        validator.huml.appendRule("",
                new String[]{ "enums" },
                "error",
                (EnumModel model) -> validator.huml.EQUALS(X.GetValue(model), model.getValue()),
                false
        );

        return validator;
    }

    public static EnumValidator createLastEnumValidator()
    {
        EnumValidator validator = new EnumValidator();
        validator.huml = new HUMLFramework();

        HUMLFramework.Variable X = validator.huml.CreateVariable("X", (EnumModel model) ->
                validator.huml.LAST(model.getEnums())

        );

        //HUMLFramework.Variable X = huml.CreateVariable("X", (BooleanModel model) -> huml.FIRST(Y.GetValue(model)));

        validator.huml.appendRule("",
                new String[]{ "enums" },
                "error",
                (EnumModel model) -> validator.huml.EQUALS(X.GetValue(model), model.getValue()),
                false
        );

        return validator;
    }

    public String getValidatorID(){
        return "";
    }

    public HUMLFramework.OpenValidationSummary validate(EnumModel model) {
        return huml.validate(model);
    }

    @Override
    public HUMLFramework.OpenValidationSummary validate(Object model) {
        return this.validate((EnumModel) model);
    }
}

File: OpenValidationFramework-CSharp/OpenValidationFramework-CSharp/Validator/EnumValidator.cs
using System;
using OpenValidationFramework_CSharp.Model;

namespace OpenValidationFramework_CSharp.Validator
{
    public class EnumValidator
    {
        public HUMLFramework huml;

        public static EnumValidator createValidatorWithMultipleArraysInVars() //TESTValidator_last_function_on_variable_containing_a_function_with_variable
        {
            EnumValidator val = new EnumValidator();

            var array = val.huml.CreateVariable("array", (EnumModel model) => model.Enums);

            var last2Array = val.huml.CreateVariable("last2Array", (EnumModel model) => val.huml.LAST(array.GetValue(model), 2.0));

            var X = val.huml.CreateVariable("X", (EnumModel model) => val.huml.LAST(last2Array.GetValue(model)));

            val.huml.appendRule("",
                new String[]{ "colors" },
                "error",
                (EnumModel model) => val.huml.EQUALS(X.GetValue(model), Color.Green),
                false
            );
            
            return val;
        }

        private EnumValidator()
        {
            this.huml = new HUMLFramework();
        }

        public String ValidatorID {
            get {
                return "";
            }
        }

        public HUMLFramework.OpenValidationSummary Validate(EnumModel model) {
            return huml.validate(model);
        }

        public HUMLFramework.OpenValidationSummary Validate(Object model) {
            return this.Validate((EnumModel)model);
        }

        
    }
}